app/src/main/java/com/cornellappdev/android/eatery/WeeklyMenuFragment.java

```diff
@@ -0,0 +1,364 @@
+		centralText = rootView.findViewById(R.id.central_header);
+
+		// Set OnClickListener for dates
+		date0 = rootView.findViewById(R.id.date0);
+		date1 = rootView.findViewById(R.id.date1);
+		date2 = rootView.findViewById(R.id.date2);
+		date3 = rootView.findViewById(R.id.date3);
+		date4 = rootView.findViewById(R.id.date4);
+		date5 = rootView.findViewById(R.id.date5);
+		date6 = rootView.findViewById(R.id.date6);
+
+		date0.setOnClickListener(this);
+		date1.setOnClickListener(this);
+		date2.setOnClickListener(this);
+		date3.setOnClickListener(this);
+		date4.setOnClickListener(this);
+		date5.setOnClickListener(this);
+		date6.setOnClickListener(this);
+
+		// Layout for menu list
+		getActivity().setTitle("Upcoming Menus");
+		DisplayMetrics dm = new DisplayMetrics();
+		getActivity().getWindowManager().getDefaultDisplay().getMetrics(dm);
+		int width = dm.widthPixels;
+		expListViewWest.setIndicatorBounds(width - 250, width);
+		expListViewNorth.setIndicatorBounds(width - 250, width);
+		expListViewCentral.setIndicatorBounds(width - 250, width);
+
+		lastExpandedPosition = -1;
+		lastClickedListView = null;
+
+		expListViewCentral.setOnGroupExpandListener(
+				new NonScrollExpandableListView.OnGroupExpandListener() {
+					@Override
+					public void onGroupExpand(int i) {
+						if (lastClickedListView != null
+								&& lastExpandedPosition != -1
+								&& i != lastExpandedPosition) {
+							lastClickedListView.collapseGroup(lastExpandedPosition);
+						}
+						lastExpandedPosition = i;
+						lastClickedListView = expListViewCentral;
+					}
+				});
+
+		expListViewWest.setOnGroupExpandListener(
+				new NonScrollExpandableListView.OnGroupExpandListener() {
+					@Override
+					public void onGroupExpand(int i) {
+						if (lastClickedListView != null
+								&& lastExpandedPosition != -1
+								&& i != lastExpandedPosition) {
+							lastClickedListView.collapseGroup(lastExpandedPosition);
+						}
+						lastExpandedPosition = i;
+						lastClickedListView = expListViewWest;
+					}
+				});
+		expListViewNorth.setOnGroupExpandListener(
+				new NonScrollExpandableListView.OnGroupExpandListener() {
+					@Override
+					public void onGroupExpand(int i) {
+						if (lastClickedListView != null
+								&& lastExpandedPosition != -1
+								&& i != lastExpandedPosition) {
+							lastClickedListView.collapseGroup(lastExpandedPosition);
+						}
+						lastExpandedPosition = i;
+						lastClickedListView = expListViewNorth;
+					}
+				});
+
+		// Populate list of date TextViews on header
+		dateList.add((TextView) rootView.findViewById(R.id.date0));
+		dateList.add((TextView) rootView.findViewById(R.id.date1));
+		dateList.add((TextView) rootView.findViewById(R.id.date2));
+		dateList.add((TextView) rootView.findViewById(R.id.date3));
+		dateList.add((TextView) rootView.findViewById(R.id.date4));
+		dateList.add((TextView) rootView.findViewById(R.id.date5));
+		dateList.add((TextView) rootView.findViewById(R.id.date6));
```

Comment:
if you convert the above snippet into a `for` loop, you could do something similar here

Reply:
i ended up eliminating this